GooglePlayInstant/Tests/Editor/QuickDeploy/LoadingScreenGeneratorTest.cs

```diff
@@ -0,0 +1,99 @@
+    /// </summary>
+    [TestFixture]
+    public class LoadingScreenGeneratorTest
+    {
+        private const string TestGameObjectName = "testing_object";
+
+        private static readonly string TestLoadingScreenJsonPath =
+            Path.Combine("Assets", LoadingScreenGenerator.LoadingScreenJsonFileName);
+
+        [Test]
+        public void TestAddLoadingScreenScript()
+        {
+            var loadingScreenGameObject = new GameObject(TestGameObjectName);
+            LoadingScreenGenerator.AddLoadingScreenScript(loadingScreenGameObject);
+            Assert.IsNotNull(loadingScreenGameObject.GetComponent<LoadingScreenScript>(),
+                "A script should be attached to the loading screen object.");
+        }
+
+        [Test]
+        public void TestAddLoadingScreenImage()
+        {
+            const string testImage = "example.png";
+
+            // Creates the file stream and disposes of it, so that adding loading screen image to scene can
+            // also read from said file.
+            using (File.Create(testImage)) ;
+
+            var loadingScreenGameObject = new GameObject(TestGameObjectName);
+
+            LoadingScreenGenerator.AddLoadingScreenImageToScene(loadingScreenGameObject, testImage);
+
+            Assert.IsNotNull(loadingScreenGameObject.GetComponent<Canvas>(),
+                "A canvas component should have been added to the loading screen game object.");
+            Assert.IsNotNull(loadingScreenGameObject.GetComponent<Image>(),
+                "An image component should have been added to the loading screen game object.");
+        }
+
+        [Test]
+        public void TestGenerateLoadingScreenConfigFileWithString()
+        {
+            const string testUrl = "test.co";
+
+            LoadingScreenGenerator.GenerateLoadingScreenConfigFile(testUrl, TestLoadingScreenJsonPath);
+
+            var loadingScreenConfigJson =
+                AssetDatabase.LoadAssetAtPath(TestLoadingScreenJsonPath, typeof(TextAsset)).ToString();
+
+            var loadingScreenConfig = JsonUtility.FromJson<LoadingScreenConfig>(loadingScreenConfigJson);
+
+            Assert.AreEqual(testUrl, loadingScreenConfig.assetBundleUrl);
+        }
+
+        [Test]
+        public void TestGenerateLoadingScreenConfigFileWithEmptyString()
+        {
+            LoadingScreenGenerator.GenerateLoadingScreenConfigFile("", TestLoadingScreenJsonPath);
+
+            var loadingScreenConfigJson =
+                AssetDatabase.LoadAssetAtPath(TestLoadingScreenJsonPath, typeof(TextAsset)).ToString();
+
+            var loadingScreenConfig = JsonUtility.FromJson<LoadingScreenConfig>(loadingScreenConfigJson);
+
+            Assert.IsEmpty(loadingScreenConfig.assetBundleUrl);
+        }
+
+        // Dispose of temporarily created file.  
+        [OneTimeTearDown]
+        public void Cleanup()
+        {
+            AssetDatabase.DeleteAsset(TestLoadingScreenJsonPath);
```

Comment:
Do you also need to delete the file at TestLoadingScreenJsonPath? After the tests are run is this file left sitting around?

Reply:
Yep, it's just sitting around.